L1Trigger/Phase2L1ParticleFlow/src/newfirmware/dataformats/egamma.h

```diff
@@ -5,44 +5,67 @@
 #include "bit_encoding.h"
 
 namespace l1ct {
-  struct EGIsoObj {
+
+  struct EGObj {
     pt_t hwPt;
-    eta_t hwEta;  // at calo face
-    phi_t hwPhi;
-    // FIXME: add quality and isolation
-    // uint16_t hwQual;
-    // uint16_t hwIso;
+    glbeta_t hwEta;  // at calo face
+    glbphi_t hwPhi;
+    quality_t hwQual;
 
     int intPt() const { return Scales::intPt(hwPt); }
     int intEta() const { return hwEta.to_int(); }
     int intPhi() const { return hwPhi.to_int(); }
+    int intQual() const { return hwQual.to_int(); }
+
     float floatPt() const { return Scales::floatPt(hwPt); }
     float floatEta() const { return Scales::floatEta(hwEta); }
     float floatPhi() const { return Scales::floatPhi(hwPhi); }
 
-    inline bool operator==(const EGIsoObj &other) const {
-      // FIXME: add quality and isolation
-      return hwPt == other.hwPt && hwEta == other.hwEta && hwPhi == other.hwPhi;
+    inline bool operator==(const EGObj &other) const {
+      return hwPt == other.hwPt && hwEta == other.hwEta && hwPhi == other.hwPhi && hwQual == other.hwQual;
     }
 
-    inline bool operator>(const EGIsoObj &other) const { return hwPt > other.hwPt; }
-    inline bool operator<(const EGIsoObj &other) const { return hwPt < other.hwPt; }
+    inline bool operator>(const EGObj &other) const { return hwPt > other.hwPt; }
+    inline bool operator<(const EGObj &other) const { return hwPt < other.hwPt; }
 
     inline void clear() {
-      // FIXME: add quality and isolation
       hwPt = 0;
       hwEta = 0;
       hwPhi = 0;
+      hwQual = 0;
     }
+  };
```

Comment:
if EGObj is used anywhere as input or output of an algorithm, add a BITWIDTH & pack()
in general it seems an overkill to have a new class just to add one member, considering that  if a datamember is never used the firmware will most likely synthethize it away